locations/spiders/teremok_ru.py

```diff
@@ -0,0 +1,54 @@
+from urllib.parse import urljoin
+
+import scrapy
+from scrapy.http import JsonRequest
+
+from locations.dict_parser import DictParser
+from locations.hours import DAYS_RU, NAMED_DAY_RANGES_RU, OpeningHours
+
+
+class TeremokRUSpider(scrapy.Spider):
+    name = "teremok_ru"
+    allowed_domains = ["teremok.ru"]
+    item_attributes = {"brand": "Теремок", "brand_wikidata": "Q4455593"}
+    start_urls = ["https://teremok.ru/"]
+
+    def parse(self, response):
+        city_selectors = response.xpath("//a[@class='b-geo__link b-link b-link--br-white b-link--thin js-city']")
+        for city in city_selectors:
+            city_id = city.xpath("@data-id").get()
+            city_name = city.xpath("text()").get()
+            yield JsonRequest(
+                urljoin(self.start_urls[0], f"/api/place/list/?city_id={city_id}"),
+                callback=self.parse_city,
+                meta={"city_name": city_name},
+            )
+
+    def parse_city(self, response):
+        city_name = response.meta["city_name"]
+        for poi in response.json()["result"]["items"]:
+            yield self.parse_poi(poi, city_name)
+
+    def parse_poi(self, poi, city_name):
+        item = DictParser.parse(poi)
+        item["lat"] = poi.get("map", [])[0]
```

Comment:
I think the point of .get is to be safe, and offer a default value for when there isn't a map value. However in this case accessing [][0] will still crash. So it's no safer than lat = poi[map][0].

The safe way is to give the array the expected length. Lat = poi.get(map [none])[0]

Assuming the count of map is always 2, we can condense these 2 lines to lat, lon = poi.get(map, [none, none])

Reply:
Thanks @Cj-Malone!